packages/@uppy/aws-s3/src/isXml.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Remove parameters like `charset=utf-8` from the end of a mime type string.
+ *
+ * @param {string} mimeType - The mime type string that may have optional parameters.
+ * @returns {string} The "base" mime type, i.e. only 'category/type'.
+ */
+function removeMimeParameters (mimeType) {
+  return mimeType.replace(/;.*$/, '')
+}
+
+/**
+ * Check if a response contains XML based on the response object and its text content.
+ *
+ * @param {string} content - The text body of the response.
+ * @param {object|XMLHttpRequest} xhr - The XHR object or response object from Companion.
```

Comment:
@goto-bus-stop This should be `@param {XMLHttpRequest}` not `@param {object|XMLHttpRequest}` because this method will fail if xhr is not of type `XMLHttpRequest` since it calls `xhr.getResponseHeader`